src/PAModel/Utility/FilePath.cs

```diff
@@ -30,6 +33,29 @@ public string ToPlatformPath()
             return Path.Combine(_pathSegments.Select(Utilities.EscapeFilename).ToArray());
         }
 
+        /// <summary>
+        /// Checks if the path length is more than MAX_PATH (260)
+        /// then it truncates the path, and adds a hash to the filename while retaining the file extension.
+        /// </summary>
+        /// <param name="path">Full path of the file.</param>
+        /// <returns></returns>
+        public static string ToValidPath(string path)
+        {
+            if (path.Length > MAX_PATH)
+            {
+                var extension = path.EndsWith(yamlExtension)
+                    ? yamlExtension
+                    : path.EndsWith(editorStateExtension)
+                    ? editorStateExtension
+                    : Path.GetExtension(path);
+
+                var hash = string.Format("{0:X}", path.GetHashCode());
```

Comment:
>path.GetHashCode() [](http://example.com/codeflow?start=50&length=18)

Danger! GetHashCode is not deterministic across process restarts. We need a deterministic hash function. since this isn't crypto, we can just grab a 3-liner and use that.